YAFC/Utils/CommandLineParser.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System.IO;
+using System;
+
+namespace YAFC {
+    public static class CommandLineParser {
+        public static string lastError { get; private set; } = string.Empty;
+        public static bool helpRequested { get; private set; }
+
+        public static bool errorOccured => !string.IsNullOrEmpty(lastError);
+
+        public static ProjectDefinition Parse(string[] args) {
+            ProjectDefinition options = new ProjectDefinition();
+
+            if (args.Length == 0) {
+                return options;
+            }
+
+            if (!args[0].StartsWith("--")) {
+                options.dataPath = args[0];
+                if (!Directory.Exists(options.dataPath)) {
+                    lastError = $"Data path '{options.dataPath}' does not exist.";
+                    return null;
+                }
+            }
+
+            for (int i = string.IsNullOrEmpty(options.dataPath) ? 0 : 1; i < args.Length; i++) {
+                switch (args[i]) {
+                    case "--mods-path":
+                        if (i + 1 < args.Length && !args[i + 1].StartsWith("--")) {
```

Comment:
> !args[i + 1].StartsWith("--")

I could name my folder `--factorio-test-mods`? I do not think there is a need for this check as it will work fine without, or check if this folder exists and only return the missing argument error when it does not.

Reply:
That is evil 😄 

But now entirely possible 😉

Reply:
It certainly is evil, sorry about that :smiling_imp:

But it was to make a point about limiting the use cases without (a good) reason. And usually, at some point there will be a user impacted and asking about it :smile:

Reply:
Don't you dare 😨 

Here would be the point where I can only tell the user: Use at your own risk, be nice, and don't use silly directory names 🤣